src/CodeEditor.tsx

```diff
@@ -49,17 +49,17 @@ export const supportedLanguages = {
   ...optionalSupport,
 };
 
-const CodeEditorView = (props: ICodeEditorProps & { className: string }) => {
+const CodeEditorView = React.forwardRef((props: ICodeEditorProps & { className: string }, ref: any) => {
```

Comment:
Let's add some typing here.
```ts
React.forwardRef<ReactSimpleCodeEditor, ICodeEditorProps & { className: string }>((props, ref) => {
```

Reply:
Why? TS seems to do a good job infering it automatically.
![image](https://user-images.githubusercontent.com/587740/49176313-92a74300-f318-11e8-9d13-37ab6223403c.png)

Edit: Ooooh I see, it _doesnt_ do a good job inferring it. `ForwardRefExoticComponent` is not as specific as `ReactSimpleCodeEditor`. OK, I got this!

Reply:
Weirdly, it seems like TS still sees the same type:
![image](https://user-images.githubusercontent.com/587740/49177342-021e3200-f31b-11e8-95c9-a61e13a2b8ea.png)

Reply:
It does not seem to be exactly the same one. See the lack of ReactSimpleCodeEditor in RefAttributes in the first screenshot.

Reply:
I'm nearsighted... I only see the first 3 lines of any TypeScript type. :)